plugins/obs-filters/noise-suppress-filter.c

```diff
@@ -158,6 +204,76 @@ static inline void alloc_channel(struct noise_suppress_data *ng,
 #endif
 #ifdef LIBRNNOISE_ENABLED
 	ng->rnn_states[channel] = rnnoise_create(NULL);
+#endif
+#ifdef LIBNVAFX_ENABLED
+	int err;
+	if (!ng->handle[0] && ng->use_nvafx && nvafx_loaded) {
+		ng->sample_rate = NVAFX_SAMPLE_RATE;
+
+		for (int i = 0; i < MAX_PREPROC_CHANNELS; i++) {
+			err = NvAFX_CreateEffect(NVAFX_EFFECT_DENOISER,
+						 &ng->handle[i]);
+			if (err != NVAFX_STATUS_SUCCESS) {
+				do_log(LOG_ERROR,
+				       "NvAFX_CreateEffect() failed, error %i",
+				       err);
+				ng->use_nvafx = false;
+			}
+			err = NvAFX_SetU32(ng->handle[i],
+					   NVAFX_PARAM_DENOISER_SAMPLE_RATE,
+					   ng->sample_rate);
+			if (err != NVAFX_STATUS_SUCCESS) {
+				do_log(LOG_ERROR,
+				       "NvAFX_SetU32(Sample Rate: %d) failed, error %i",
+				       ng->sample_rate, err);
+				ng->use_nvafx = false;
+			}
+			err = NvAFX_SetString(ng->handle[i],
+					      NVAFX_PARAM_DENOISER_MODEL_PATH,
+					      ng->model);
+			if (err != NVAFX_STATUS_SUCCESS) {
+				do_log(LOG_ERROR,
+				       "NvAFX_SetString() failed, error %i",
+				       err);
+				ng->use_nvafx = false;
+			}
+			err = NvAFX_Load(ng->handle[i]);
+			if (err != NVAFX_STATUS_SUCCESS) {
+				do_log(LOG_ERROR,
+				       "NvAFX_Load() failed with error %i",
+				       err);
+				ng->use_nvafx = false;
+			}
+		}
+		if (ng->use_nvafx) {
+			err = NvAFX_GetU32(ng->handle[0],
+					   NVAFX_PARAM_DENOISER_NUM_CHANNELS,
+					   &ng->num_channels);
+			if (err != NVAFX_STATUS_SUCCESS) {
+				do_log(LOG_ERROR,
+				       "NvAFX_GetU32() failed to get the number of channels, error %i",
+				       err);
+				ng->use_nvafx = false;
+			}
+			if (ng->num_channels != 1)
+				do_log(LOG_ERROR,
+				       "The number of channels is not 1 in the sdk any more ==> update code");
```

Comment:
This appears to be an error, but it doesn't mark ng->use_nvafx to false.

Reply:
right, fixed in the new code pushed.